### PR TITLE
Fix problem with the CLI never exiting

### LINKED ISSUE #53: Starting linter on a remote resource will block indefinitely
The following never stops 

```
awesome-lint 'https://gist.githubusercontent.com/maximveksler/e04db4c4e50b5756f8865a836028bacf/raw/6fdc20ef697e2e2445e1c52a14012c1f32797344/README.md'
⠙ Linting
```

### LINKED ISSUE #68: Log error when trying to lint non-existent file
When trying to run `awesome-lint` on a file that doesn't exist (e.g. `awesome-lint unicorn.md`), no error is being reported. Instead, the command runs indefinitely until you `Control+C` it and see the file path was invalid in the first place.

### PR DESCRIPTION
**Pull request by chinesedfan**
Fixes #53 and fixes #68.

### CODE REVIEW

**Comment by transitive-bullshit** [comment]
I think we want to support other git repos in addition to GitHub.

**Comment by itaisteinherz** [comment]
We should add some tests to make sure this works (and continues to work) as expected.

**Comment by chinesedfan** [comment]
@transitive-bullshit Maybe in future. See https://github.com/sindresorhus/awesome-lint/issues/53#issuecomment-477079741.

@itaisteinherz Right. After the scope is confirmed, I will add tests.

**Comment by transitive-bullshit** [comment]
@chinesedfan sounds good 😄

File: index.js [hunk 1]
```diff
@@ -34,9 +36,23 @@ const lint = options => {
...
+		spinner.fail(error.message);
+		process.exitCode = 1;
+	}
+};
+
+lint._report = async (options, spinner) => {
 	let temp = null;
 
 	if (isUrl(options.filename)) {
+		if (!rGithubRepo.test(options.filename)) {
```
Comment (itaisteinherz) [inline index.js]: [`is-github-url`](https://github.com/alferov/is-github-url) should probably be used here instead, as it's properly tested and can handle different URL variations and structures.

Comment (chinesedfan) [reply index.js]: @itaisteinherz Great! I was wondering there had a package like that.

File: test/cli.js [hunk 2]
```diff
@@ -7,3 +7,17 @@ test('main', async t => {
 		/Missing Awesome badge/
 	);
 });
+
+test('main - non-exsitent file', async t => {
```
Comment (sindresorhus) [inline test/cli.js]: Typo

File: test/cli.js [hunk 3]
```diff
@@ -7,3 +7,17 @@ test('main', async t => {
...
+	await t.throwsAsync(
+		execa.stderr('./cli.js', ['test/fixtures/non-exsitent.md']),
+		/Couldn't find the file/
+	);
+});
+
+test('main - invalid Github repository', async t => {
```
Comment (sindresorhus) [inline test/cli.js]: Typo

File: index.js [hunk 4]
```diff
@@ -34,9 +35,23 @@ const lint = options => {
...
+		process.exitCode = 1;
+	}
+};
+
+lint._report = async (options, spinner) => {
 	let temp = null;
 
 	if (isUrl(options.filename)) {
+		if (!isGithubUrl(options.filename, {repository: true})) {
+			throw new Error(`Invalid Github repository url: ${options.filename}`);
```
Comment (sindresorhus) [inline index.js]:
```suggestion
			throw new Error(`Invalid GitHub repo URL: ${options.filename}`);
```

**Comment by chinesedfan** [comment]
@sindresorhus Updated. Thanks for your reviews and feel shame for my silly mistakes.

File: test/cli.js [hunk 5]
```diff
@@ -7,3 +7,17 @@ test('main', async t => {
 		/Missing Awesome badge/
 	);
 });
+
+test('main - non-existent file', async t => {
+	await t.throwsAsync(
+		execa.stderr('./cli.js', ['test/fixtures/non-exsitent.md']),
```
Comment (transitive-bullshit) [inline test/cli.js]: can you please rename this file to `non-existent.md`? just a small spelling mistake

**Comment by transitive-bullshit** [comment]
> Thanks for your reviews and feel shame for my silly mistakes.

No reason to feel any shame -- we all make mistakes 😀

Thanks a lot for your contributions so far! I think just fixing one small typo and this PR is good to go.

File: test/cli.js (hunk 5 above)
Comment (chinesedfan) [reply test/cli.js]: @transitive-bullshit Done.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by chinesedfan** [comment]
@sindresorhus Conflict resolved.

**Pull request merged**